IBAnimatable/CornerDesignable.swift

```diff
@@ -17,43 +17,75 @@ public protocol CornerDesignable {
   var cornerSides: CornerSides { get set }
 }
 
-public extension CornerDesignable where Self: UIView {
-  public func configureCornerRadius() {
-    if !cornerRadius.isNaN && cornerRadius > 0 {
-      if cornerSides == .AllSides {
-        layer.cornerRadius = cornerRadius
-      } else {
-        layer.cornerRadius = 0.0
+public extension CornerDesignable where Self: UICollectionViewCell {
+    public func configureCornerRadius() {
+        if !cornerRadius.isNaN && cornerRadius > 0 {
 
-        // if a layer mask is specified, remove it
-        layer.mask?.removeFromSuperlayer()
+            // Remove any previous corner mask, i.e. coming from UIView type implementation
+            if layer.mask?.name == "cornerSideLayer" {
+                layer.mask?.removeFromSuperlayer()
+            }
+            layer.cornerRadius = 0.0
+            
+            if cornerSides == .AllSides {
+                contentView.layer.cornerRadius = cornerRadius
+            } else {
+                contentView.layer.cornerRadius = 0.0
+                
+                // if a layer mask is specified, remove it
+                contentView.layer.mask?.removeFromSuperlayer()
 
-        // configure corner for specified sides
-        let corner = CAShapeLayer()
-        corner.name = "cornerSideLayer"
-        corner.frame = CGRect(origin: .zero, size: bounds.size)
+                contentView.layer.mask = cornerSidesLayer()
+            }
+            
+            contentView.layer.masksToBounds = true
+        } else {
+            contentView.layer.masksToBounds = false
+        }
+    }
+}
 
+public extension CornerDesignable where Self: UIView {
+    public func configureCornerRadius() {
+        if !cornerRadius.isNaN && cornerRadius > 0 {
+            if cornerSides == .AllSides {
+                layer.cornerRadius = cornerRadius
+            } else {
+                layer.cornerRadius = 0.0
+                
+                // if a layer mask is specified, remove it
+                layer.mask?.removeFromSuperlayer()
+                
+                layer.mask = cornerSidesLayer()
+            }
+        }
+    }
+    
+    fileprivate func cornerSidesLayer() -> CAShapeLayer {
+        let cornerSideLayer = CAShapeLayer()
+        cornerSideLayer.name = "cornerSideLayer"
+        cornerSideLayer.frame = CGRect(origin: .zero, size: bounds.size)
+        
         let cornerRadii = CGSize(width: cornerRadius, height: cornerRadius)
-
+        
         var roundingCorners: UIRectCorner = []
         if cornerSides.contains(.topLeft) {
-          roundingCorners.insert(.topLeft)
+            roundingCorners.insert(.topLeft)
```

Comment:
Sorry, I started to use two space a tab because most of the time, I use Mac book only (with 15-inch screen) on the train or on the couch 😂, That gives me more space to read. Can we stick with this rule for this project?